clang/include/clang/CConv/ConstraintsGraph.h

```diff
@@ -14,15 +14,17 @@
 
 #include <boost/graph/graph_traits.hpp>
 #include <boost/graph/adjacency_list.hpp>
-#include <boost/graph/dijkstra_shortest_paths.hpp>
 #include "clang/CConv/Constraints.h"
 
 using namespace boost;
 using namespace std;
 
+enum EdgeType { Ptype, Checked };
+const std::string EdgeTypeColors[2] = { "blue", "red" };
+
 class ConstraintsGraph {
 public:
-  typedef adjacency_list<setS, vecS, bidirectionalS, Atom*> DirectedGraphType;
+  typedef adjacency_list<vecS, vecS, bidirectionalS, Atom*, EdgeType> DirectedGraphType;
```

Comment:
Changed the edge collection from set to vector so that there can be duplicate edges: one for a ptyp constraint and one for a checked constraint.

Reply:
But why should this be in the graphs we use for solving things? For those graphs, they are indeed separate. Is it because you have to construct a new graph by combining the other two, so you can visualize it?